### PR TITLE
Feature/location data display refactoring

### PR DESCRIPTION
**Pull request by pcmueller**
### Is this a fix or a feature?
- Feature

### What is the change or fix?
- Refactored _top-artists.scss_ styling so that we now have a "artist card" grid display instead of a list of names.
- Refactored _TopArtists.js_ functionality so that we now create and display X number of "artist card" elements based on the number included in our `filteredArtists()` function.
- Refactored `apiCalls.getArtistImage()` function so that it now dynamically fetches a MovieBrainz artist object dynamically based on whatever ID is passed in.
- Built a `retrieveArtistImage()` function (_App.js_) to invoke the API call and set state with the returned data.  We night not actually need this, since the image property will just get assigned to each element in the `this.state.topArtists` array.  Could just make the API call right in _TopArtists.js_, with that in mind.  Probably something to figure out when we get into state management with Redux.  

### How should this be tested?
- Display in browser.  But hold off on running the commented-out code block until we've figure out the CORS issue!

EDIT: Also added / pushed up new "random country" functionality: now when you load or refresh the page, a different country will get assigned to `this.state.location`, and will be used to make TopArtists and TopTracks fetch requests.  I did this using a new `countries.js` dataset I put in the utilities directory, and using a few new functions in _utils.js_ and _App.js_.  Not all countries in the array have Last.fm data, so if the page ever fails to load, please check the console to see which country it was trying to fetch (should appear within an error or log), and then comment-out that country name in the _countries.js_ dataset.  I caught a handful of them, but any more page refreshes and I would've induced a seizure.  Let's just cut them out one at a time.

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ class App extends Component {
     this.state = {
       topArtists: [],
       topTracks: [],
-      selectedArtist: '',
+      selectedArtistID: '',
```
Comment (battan40) [inline src/components/App/App.js:18]: Much clearer data distinction

File: src/components/TopArtists/TopArtists.js [hunk 2]
```diff
@@ -9,25 +9,43 @@ const TopArtists = ({topArtists}) => {
     return topTen;
   }, []);
 
+  /****** DON'T RUN BLOCK UNTIL WE'VE FIGURED OUT 'CORS' ISSUE ******/
```
Comment (battan40) [inline src/components/TopArtists/TopArtists.js:12]: Thank you for the note.   Have we all researched the expiration date of our api key?

File: src/components/TopArtists/TopArtists.js [hunk 3]
```diff
@@ -9,25 +9,43 @@ const TopArtists = ({topArtists}) => {
...
       <h3> Top Artists in *location* </h3>
       <div className='artists-list'>
         {artistCards}
       </div>
-    </article>
+    </section>
   )
 }
 
+/*
```
Comment (battan40) [inline src/components/TopArtists/TopArtists.js:41]: Thank you for this level of specificity.

**Review by battan40 — APPROVED** [review]

**Comment by battan40** [comment]
I'm pulling this down to clear the merge conflict from minor testing updates.  However, I will hold the integrity of both of your code contributions.

File: src/components/App/App.js [hunk 4]
```diff
@@ -10,22 +12,32 @@ class App extends Component {
...
-    this.retrieveTopArtists();
-    this.retrieveTopTracks();
+    console.log(this.state.location);
+
+    this.retrieveTopArtists(this.state.location.string);
+    this.retrieveTopTracks(this.state.location.string);
+  }
+
+  getRandomLocation = () => {
+    const randomCountry= utils.getRandomElement(countries)
```
Comment (bhohnco) [inline src/components/App/App.js:33]: Great work here

File: src/components/App/App.js [hunk 5]
```diff
@@ -66,8 +93,8 @@ class App extends Component {
           <Form />
           <main className='main-section'>
             <section className='location-display'>
-              <TopArtists topArtists={this.state.topArtists}/>
-              <TopTracks topTracks={this.state.topTracks}/>
+              <TopArtists location={this.state.location.name} topArtists={this.state.topArtists} retrieveArtistImage={this.retrieveArtistImage}/>
+              <TopTracks location={this.state.location.name} topTracks={this.state.topTracks}/>
```
Comment (bhohnco) [inline src/components/App/App.js:97]: Nice!

**Review by bhohnco — APPROVED** [review]

**Pull request merged**